mvs/common/adapter_base.py

```diff
@@ -0,0 +1,146 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from pathlib import Path
+import shutil
+import os
+from common.sparse_reconstruction.preprocessing import get_intrinsics_from_report
+from common.sparse_reconstruction.feature_extractor import FeatureExtractor
+from common.sparse_reconstruction.feature_matcher import FeatureMatcher
+from common.sparse_reconstruction.sparse_reconstructor import SparseReconstructor
+from common.sparse_reconstruction.undistorter import Undistorter
+
+
+class AdapterBase(ABC):
+    def __init__(self, input_dir: Path, output_dir: Path):
+        self.input_dir = input_dir
+        self.output_dir = output_dir
+    
+    def run(self):
+        self.preprocess()
+        self._sparse_reconstruction()
+        self._convert_colmap_to_mvsnet()
+        self._load_model()
+        self._process()     
+        self._postprocess()
+
+    def preprocess(self):
+        workdir = Path(self.input_dir)
+
+        path_to_raw_images = workdir / "images_raw"
+        camera_a308_report = workdir / "Detailed Report A308.txt"
+        camera_a311_report = workdir / "Detailed Report A311.txt"
+
+        path_to_preprocessed = workdir / "images"
+        if os.path.exists(path_to_preprocessed):
+            shutil.rmtree(path_to_preprocessed)
+        path_to_preprocessed.mkdir(exist_ok=False)          
+        print(path_to_preprocessed)
+        intrinsics = dict()
+        intrinsics["a308_0"], intrinsics["a308_1"] = get_intrinsics_from_report(
+            camera_a308_report
+        )
+        intrinsics["a311_0"], intrinsics["a311_1"] = get_intrinsics_from_report(
+            camera_a311_report
+        )
+
+        for camera, intrinsic in intrinsics.items():
+            subfolder = path_to_preprocessed / camera
+            subfolder.mkdir()
+
+            with open(subfolder / "intrinsic.txt", "w") as intrinsics_file:
+                intrinsics_file.write(
+                    f"{intrinsic.f} {intrinsic.cx} {intrinsic.cy} {intrinsic.k}"
+                )
+
+        raw_images = path_to_raw_images.iterdir()
+        for raw_image in raw_images:
+            raw_image_name = raw_image.name
+
+            match len(raw_image_name), int(raw_image_name[-5]):
+                case 16, 0:
+                    chosen_camera = "a308_0"
+                case 16, 1:
+                    chosen_camera = "a308_1"
+                case 21, 0:
+                    chosen_camera = "a311_0"
+                case 21, 1:
+                    chosen_camera = "a311_1"
+                case _:
+                    raise ValueError("Invalid image name")
```

Comment:
It is better to explicitly indicate in the error message which name caused the exception